### PR TITLE
nixos/containers: Fix shellcheck issues

### PR DESCRIPTION
**Pull request by r-vdp**
## Things done

Fix issues discovered with #311394.

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/virtualisation/nixos-containers.nix [hunk 1]
```diff
@@ -109,44 +109,46 @@ let
...
         IFS=$OIFS
       fi
 
       if [ -n "$HOST_BRIDGE" ]; then
-        extraFlags+=" --network-bridge=$HOST_BRIDGE"
+        extraFlags+=("--network-bridge=$HOST_BRIDGE")
       fi
 
-      extraFlags+=" ${concatStringsSep " " (mapAttrsToList nspawnExtraVethArgs cfg.extraVeths)}"
+      ${concatStringsSep "\n" (mapAttrsToList (name: cfg: "extraFlags+=(${nspawnExtraVethArgs name cfg})") cfg.extraVeths)}
```
Comment (K900) [inline nixos/modules/virtualisation/nixos-containers.nix:137]: `lib.escapeShellArgs` here?

File: nixos/modules/virtualisation/nixos-containers.nix [hunk 2]
```diff
@@ -157,9 +159,11 @@ let
       # Kill signal handling means systemd-nspawn will pass a system-halt signal
       # to the container systemd when it receives SIGTERM for container shutdown;
       # containerInit and stage2 have to handle this as well.
+      # TODO: fix shellcheck issue properly
+      # shellcheck disable=SC2086
```
Comment (K900) [inline nixos/modules/virtualisation/nixos-containers.nix:163]: Yeah this is the fun part isn't it.

**Comment by r-vdp** [comment]
@ofborg test containers-bridge containers-custom-pkgs.nix containers-ephemeral containers-extra_veth containers-hosts containers-imperative containers-ip containers-macvlans containers-names containers-nested containers-physical_interfaces containers-portforward containers-reloadable containers-require-bind-mounts containers-restart_networking containers-tmpfs containers-unified-hierarchy

File: nixos/modules/virtualisation/nixos-containers.nix (hunk 2 above)
Comment (r-vdp) [reply nixos/modules/virtualisation/nixos-containers.nix:163]: Yeah, I still want to fix it properly, but that will take more effort.

File: nixos/modules/virtualisation/nixos-containers.nix (hunk 1 above)
Comment (r-vdp) [reply nixos/modules/virtualisation/nixos-containers.nix:137]: Running the tests with that

Comment (r-vdp) [reply nixos/modules/virtualisation/nixos-containers.nix]: Done!

**Comment by r-vdp** [comment]
The only container test failing now is containers-restart_networking, but that one also fails on master.

**Comment by r-vdp** [comment]
https://github.com/ofborg test containers-bridge containers-custom-pkgs.nix containers-ephemeral containers-extra_veth containers-hosts containers-imperative containers-ip containers-macvlans containers-names containers-nested containers-physical_interfaces containers-portforward containers-reloadable containers-require-bind-mounts containers-tmpfs containers-unified-hierarchy

**Review by K900 — APPROVED** [review]

**Pull request merged**